### PR TITLE
Bug/e UI 2030 duplicate assignees

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/EXUI-2030

### Change description ###
Merge all duplicates logically and ensure only one caseworker for each duplicate is returned

### CODE REVIEW

**Review by OgunyemiO — APPROVED** [review]

**Review by Josh-HMCTS — APPROVED** [review]

File: api/workAllocation/caseWorkerUserDataCacheService.ts [hunk 1]
```diff
@@ -171,6 +171,104 @@ export function timestampExists(): boolean {
   return !!timestamp;
 }
 
+// This ensures there are no duplicates
+export function getUniqueUsersFromResponse(userResponse: StaffUserDetails[]): StaffUserDetails[] {
+  const userIdList = [];
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply api/workAllocation/caseWorkerUserDataCacheService.ts:176]: Will change

File: api/workAllocation/caseWorkerUserDataCacheService.ts [hunk 2]
```diff
@@ -171,6 +171,104 @@ export function timestampExists(): boolean {
...
+  const uniqueUsers = [];
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply api/workAllocation/caseWorkerUserDataCacheService.ts:177]: Will change

File: api/workAllocation/caseWorkerUserDataCacheService.ts [hunk 3]
```diff
@@ -171,6 +171,104 @@ export function timestampExists(): boolean {
...
+            }
+          }
+        }
+      }
+      // user has all related services and combined locations
+      // note: non-combined locations have services undefined
+      thisUser.ccd_service_names = userServices;
+      memberProfile.base_location = baseLocationList?.length > 0 ? baseLocationList : memberProfile.base_location;
+      // ensure user is not reused
+      uniqueUsers.push(thisUser);
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply api/workAllocation/caseWorkerUserDataCacheService.ts:217]: See comment below

File: api/workAllocation/caseWorkerUserDataCacheService.ts [hunk 4]
```diff
@@ -171,6 +171,104 @@ export function timestampExists(): boolean {
...
+      userIdList.push(thisUser.staff_profile.id);
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply api/workAllocation/caseWorkerUserDataCacheService.ts:218]: So this code is within an if check that will only be true if the id for the user is not part of the userIdList of which is only added to when the if check is true. Therefore no users are duplicated and I have tested this thoroughly and it works fine. I agree we could use a set and that is the change I could make based on this comment. Do you want me to make that change?

**Review by andywilkinshmcts — DISMISSED** [review]
Would be a little more efficient to use a set for userIdList

**Review by andywilkinshmcts — APPROVED** [review]

**Review by RiteshHMCTS — APPROVED** [review]

**Pull request merged**